ocs_ci/ocs/platform_nodes.py

```diff
@@ -1272,3 +1306,137 @@ def get_worker_resource_id(self, resource):
 
         """
         return resource['StackResourceSummaries'][0]['PhysicalResourceId']
+
+
+class VSPHEREUPINode(VMWareNodes):
+    """
+    Node object representing VMWARE UPI nodes
+    """
+    def __init__(self, node_conf, node_type, compute_count):
+        """
+        Initialize necessary variables
+
+        Args:
+            node_conf (dict): of node configuration
+            node_type (str): type of node to be created RHCOS/RHEL
+            compute_count (int): number of nodes to add to existing cluster
+
+        """
+        super(VSPHEREUPINode, self).__init__()
+        self.node_conf = node_conf
+        self.node_type = node_type
+        self.compute_count = compute_count
+        self.current_compute_count = config.ENV_DATA.get('worker_replicas', 3)
```

Comment:
better to fetch the compute nodes count dynamically from the cluster, like with node.get_typed_nodes()

Reply:
Done